File: sub-mysql-web-entity/src/main/java/org/github/mysql/web/base/enums/Sex.java
package org.github.mysql.web.base.enums;

import lombok.NonNull;
import org.github.base.IEnum;

public enum Sex implements IEnum<Integer,String> {
  man(0, "男"),
  woman(1, "女");

  private final Integer code;

  private final String value;

  Sex(Integer code, String value) {
    this.code = code;
    this.value = value;
  }

  @Override
  public @NonNull Integer getCode() {
    return code;
  }

  @Override
  public @NonNull String getValue() {
    return value;
  }
}

File: sub-mysql-mydb-entity/src/main/java/org/github/mysql/mydb/base/dict/Sex.kt
package org.github.mysql.mydb.base.dict

import org.github.base.BaseEnum

enum class Sex(override val code: String, override val value: String): BaseEnum {
    /** 保密，默认值 */
  SECRET("secret","保密"),
    /** 男 */
  MAN("man","男"),
    /** 女 */
  WOMAN("woman","女"),
}
